libraries/AP_GPS/AP_GPS.h

```diff
@@ -39,6 +38,8 @@ class AP_GPS_Backend;
 /// GPS driver main class
 class AP_GPS
 {
+    friend class AP_GPS_Backend; // for access to detect_state
```

Comment:
I know this is the smallest change possible, but we really should do like in other libraries and pass the frontend reference/pointer to the backends